test/testVegaWrapper.js

```diff
@@ -417,3 +418,351 @@ describe('vegaWrapper', function() {
+        });
+
+        it('wikiraw', function () {
+            // wikiraw allows sub-domains
+            fail({ type: 'wikiraw', wiki: 'sec.org' });
+            fail({ type: 'wikiraw', wiki: 'sec.org', a: 10 });
+            fail({ type: 'wikiraw', wiki: 'asec.org', title: 'aaa' });
+            fail({ type: 'wikiraw', title: 'abc\x1Fxyz' });
+            fail({ type: 'wikiraw', title: 'abc|xyz' });
+            fail({ type: 'wikiraw', wiki: 'sec.org', title: 'abc|xyz' });
+            fail({ type: 'wikiraw', wiki: 'sec.org', title: '\x1Fxyz' });
+            passWithCors({ type: 'wikiraw', title: 'abc' }, 'https://domain.sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=abc');
+            passWithCors({ type: 'wikiraw', title: 'abc/xyz' }, 'https://domain.sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=abc%2Fxyz');
+            passWithCors({ type: 'wikiraw', wiki: 'sec.org', title: 'aaa' }, 'https://sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=aaa');
+            passWithCors({ type: 'wikiraw', wiki: 'sec.org', title: 'aaa', a: 10 }, 'https://sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=aaa');
+            passWithCors({ type: 'wikiraw', wiki: 'sec.org', title: 'abc/def' }, 'https://sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=abc%2Fdef');
+            passWithCors({ type: 'wikiraw', wiki: 'sec', title: 'aaa' }, 'https://sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=aaa');
+            passWithCors({ type: 'wikiraw', wiki: 'sec', title: 'abc/def' }, 'https://sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=abc%2Fdef');
+            passWithCors({ type: 'wikiraw', wiki: 'wikiraw.sec.org', title: 'abc' }, 'https://wikiraw.sec.org/w/api.php?format=json&formatversion=2&action=query&prop=revisions&rvprop=content&titles=abc');
+        });
+
+        it('wikifile', function () {
+            fail({ type: 'wikifile', title: 'this|pic' });
+            fail({ type: 'wikifile', title: '\x1Fimg' });
+            pass({ type: 'wikifile', title: 'Einstein_1921.jpg' }, 'https://domain.sec.org/wiki/Special:Redirect/file/Einstein_1921.jpg');
+            pass({ type: 'wikifile', title: 'Einstein_1921.jpg', width: 10 }, 'https://domain.sec.org/wiki/Special:Redirect/file/Einstein_1921.jpg?width=10');
+        });
+
+        it('wikidatasparql', function () {
+            fail({ type: 'wikidatasparql'});
+            fail({ type: 'wikidatasparql', path: 'a' });
+            fail({ type: 'wikidatasparql', a: 10 });
+            fail({ type: 'wikidatasparql', aquery: 1 });
+            pass({ type: 'wikidatasparql', query: 1 }, 'http://wikidatasparql.nonsec.org/bigdata/namespace/wdq/sparql?query=1');
```

Comment:
this one should fail -- query must be a string.  Make sure only string passes  for query param.